clients/rest-client-core/src/main/java/uk/gov/justice/services/clients/core/RestClientProcessor.java

```diff
@@ -91,6 +91,10 @@ private JsonObject addMetadataIfMissing(final JsonObject responseAsJsonObject, f
             return responseAsJsonObject;
         }
 
+        if (cppId == null) {
+            throw new RuntimeException(format("%s is required in request header", CPPID));
```

Comment:
I discussed this with Andrew and he said that we should return 400 without any body.